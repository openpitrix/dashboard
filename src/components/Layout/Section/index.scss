.section {
  display: flex;
  margin-bottom: 24px;
  width: 100%;
  height: auto;
}
