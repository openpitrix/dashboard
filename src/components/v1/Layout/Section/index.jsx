import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './index.scss';

const Section = ({ size, className, children }) => {
  return <div className={classnames(styles.section, className)}>{children}</div>;
};

Section.propTypes = {
  size: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};

Section.defaultProps = {
  size: 4 // col span
};

export default Section;
