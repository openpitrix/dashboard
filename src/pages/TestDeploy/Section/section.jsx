import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import _ from 'lodash';

import { Radio, Input, Select, Slider } from 'components/Base';

import styles from './index.scss';

export default class Section extends React.Component {
  static propTypes = {
    detail: PropTypes.shape({
      defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      description: PropTypes.string,
      keyName: PropTypes.string.isRequired, // key is reserved by react
      label: PropTypes.string,
      required: PropTypes.bool,
      type: PropTypes.string.isRequired,
      renderType: PropTypes.string,
      items: PropTypes.array
    }),
    className: PropTypes.string
  };

  static defaultProps = {
    detail: {
      defaultValue: '',
      description: '',
      keyName: '',
      label: '',
      required: false,
      type: 'string'
    }
  };

  handleChange = val => {
    const { keyName, renderType } = this.props.detail;

    if (typeof val === 'object' && val.nativeEvent) {
      // react event object
      val = val.currentTarget.value;
    }

    // todo: unify onChange
    console.log(`change ${renderType}- ${keyName}: `, val);
  };

  renderLabel() {
    const { keyName, label, renderType } = this.props.detail;

    return (
      <label
        className={classnames(styles.label, {
          [styles.isRadio]: renderType === 'radio'
        })}
      >
        {_.capitalize(label || keyName)}
      </label>
    );
  }

  renderItem() {
    const {
      renderType,
      description,
      defaultValue,
      keyName,
      items,
      required,
      ...rest
    } = this.props.detail;

    let content = null;

    switch (renderType) {
      case 'input':
        content = (
          <Fragment>
            <Input
              className={styles.input}
              name={keyName}
              type="text"
              defaultValue={defaultValue}
              maxLength={50}
              onChange={this.handleChange}
              required={required}
            />
            <p className={styles.tips}>{description}</p>
          </Fragment>
        );
        break;

      case 'number':
        content = (
          <Fragment>
            <Input
              className={styles.input}
              name={keyName}
              type="number"
              defaultValue={defaultValue}
              onChange={this.handleChange}
              min={rest.min}
              max={rest.max}
              required={required}
            />
            <p className={styles.tips}>
              Range: {rest.min} - {rest.max}
            </p>
          </Fragment>
        );
        break;

      case 'radio':
        content = (
          <Radio.Group
            className={styles.radio}
            value={defaultValue === undefined ? rest.range[0] : defaultValue}
            onChange={this.handleChange}
            name={keyName}
          >
            {rest.range.map((item, idx) => (
              <Radio value={item} key={idx}>
                {item}
              </Radio>
            ))}
          </Radio.Group>
        );
        break;

      case 'select':
        // todo
        content = (
          <Select
            className={styles.select}
            value={defaultValue}
            onChange={this.handleChange}
            disabled={items.length === 0}
          >
            {items.map(({ version_id, name }) => (
              <Select.Option key={version_id} value={version_id}>
                {name}
              </Select.Option>
            ))}
          </Select>
        );
        break;

      case 'text':
        content = (
          <textarea
            className={styles.text}
            name={keyName}
            defaultValue={defaultValue}
            maxLength={1000}
            onChange={this.handleChange}
          />
        );
        break;

      case 'slider':
        content = (
          <Fragment>
            <Slider
              min={rest.min}
              max={rest.max}
              step={rest.step}
              defaultValue={defaultValue}
              onChange={this.handleChange}
              className={styles.slider}
            />
            <span className={styles.sliderInput}>
              <Input
                className={styles.inputSmall}
                type="number"
                name={keyName}
                defaultValue={defaultValue}
                min={rest.min}
                max={rest.max}
                onChange={this.handleChange}
              />{' '}
              GB
            </span>
            <p className={styles.tips}>
              {`${rest.min}GB - ${rest.max}GB, The volume size for each instance`}
            </p>
          </Fragment>
        );
        break;

      default:
        break;
    }

    return content;
  }

  render() {
    const { className } = this.props;

    return (
      <div className={classnames(styles.row, className)}>
        {this.renderLabel()}
        <div className={styles.item}>{this.renderItem()}</div>
      </div>
    );
  }
}
