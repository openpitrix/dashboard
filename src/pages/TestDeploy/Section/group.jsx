import React from 'react';
import PropTypes from 'prop-types';

import Section from './section';
import factory from 'lib/config-parser/section';

import styles from './index.scss';

export default class DeployGroup extends React.Component {
  static propTypes = {
    detail: PropTypes.object,
    seq: PropTypes.number
  };

  static defaultProps = {
    detail: {},
    seq: 0
  };

  renderSingle = item => {
    if (typeof item.toJSON === 'function') {
      item = item.toJSON();
    }

    return <Section detail={item} />;
  };

  render() {
    const { detail, seq } = this.props;
    const isNormalGroup = Array.isArray(detail.items);
    const title = isNormalGroup ? detail.title : detail.label || detail.key;
    const items = isNormalGroup ? detail.items : factory(detail.properties);

    return (
      <div className={styles.group}>
        <h2 className={styles.title}>{`${seq + 1}. ${title}`}</h2>
        {items.map((item, idx) => {
          return <div key={idx}>{this.renderSingle(item)}</div>;
        })}
      </div>
    );
  }
}
