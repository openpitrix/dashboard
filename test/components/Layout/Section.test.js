import Section from 'components/Layout/Grid/Section';

describe('Layout/Section', () => {
  it('basic render', () => {
    const wrapper = render(
      <Section className="test" size={8} offset={4}>
        <div>test</div>
      </Section>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
