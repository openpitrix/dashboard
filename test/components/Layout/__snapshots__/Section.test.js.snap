// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/Section basic render 1`] = `
<div
  class="section section-size-8 section-offset-4} test"
>
  <div>
    test
  </div>
</div>
`;
